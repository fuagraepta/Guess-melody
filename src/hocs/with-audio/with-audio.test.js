import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';

import withAudio from './with-audio.js';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

describe(`The component with audio is rendered correctly`, () => {
  it(`With Audio correctly renders`, () => {
    const MockComponentWrapped = withAudio(MockComponent);

    const mockComponentWrapped = renderer
    .create(<MockComponentWrapped
      isLoading = {false}
      isPlaying = {false}
      src = {``}
      onPlayButtonClick = {jest.fn()}
    />)
      .toJSON();

    expect(mockComponentWrapped).toMatchSnapshot();
  });
});
