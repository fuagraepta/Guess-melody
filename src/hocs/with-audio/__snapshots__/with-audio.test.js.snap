// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component with audio is rendered correctly With Audio correctly renders 1`] = `
<div>
  <audio />
</div>
`;
